odo/convert.py

```diff
@@ -213,14 +219,30 @@ def _():
             yield convert(pd.DataFrame, [], **kwargs)
     else:
         df = convert(pd.DataFrame, first, **kwargs)
+        df1, n1 = mkindex(df, 0)
 
         def _():
-            yield df
+            n = n1
+            yield df1
             for i in rest:
-                yield convert(pd.DataFrame, i, **kwargs)
+                df = convert(pd.DataFrame, i, **kwargs)
+                df, n = mkindex(df, n)
+                yield df
     return chunks(pd.DataFrame)(_)
 
 
+def _ignore_index(df, start):
+    return df, start
+
+
+def _add_index(df, start, _idx_type=getattr(pd, 'RangeIndex',
+                                            compose(pd.Index, np.arange))):
+    stop = start + len(df)
+    idx = pd.Index(_idx_type(start=start, stop=stop))
```

Comment:
I don't think we still need to wrap this in Index because the result of `_idx_type` should be an index. Other that this lgtm